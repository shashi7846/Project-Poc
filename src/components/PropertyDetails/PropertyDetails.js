import React, { useState, useEffect } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faAreaChart,
  faQuestion,
  faMapMarker,
  faMapPin,
  faCircleInfo,
} from "@fortawesome/free-solid-svg-icons";
import { Link, useNavigate } from "react-router-dom";
import { PostPropertyDetails } from "../../api/Api";
import { toast } from "react-toastify";
import { validatePropertyDetailsForm } from "../validators";
import { PuffLoader } from "react-spinners";
import { OverlayTrigger } from "react-bootstrap";

import Popover from "react-bootstrap/Popover";

export const customPopover = (
  <Popover id="popover-basic">
    <Popover.Header as="h3">Address should only contain..</Popover.Header>
    <Popover.Body>
      <div>1). Alphabets Upper/Lower case</div>
      <div>2). " - "</div>
      <div>3). " / "</div>
      <div>4). Numbers 0-9</div>
      <div>{"->"} Anything other than this is not acceptable.</div>
    </Popover.Body>
  </Popover>
);

const defaultData = {
  address: "",
  area: "",
  pincode: "",
  purposeOfLoan: "",
};

function PropertyDetails(props) {
  const [propertyDetails, setPropertyDetails] = useState(defaultData);

  const [loading, setLoading] = useState(false);
  useEffect(() => {
    setLoading(true);
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, []);

  let navigate = useNavigate();
  const handleChange = (e) =>
    setPropertyDetails({ ...propertyDetails, [e.target.name]: e.target.value });

  const handleOnSubmit = async (e) => {
    try {
      e.preventDefault();
      const errors = validatePropertyDetailsForm(propertyDetails);
      if (errors.length) {
        throw new Error(errors[0]);
      }
      await PostPropertyDetails(localStorage.getItem("id"), {
        propertyDetails,
      });
      setPropertyDetails(defaultData);
      navigate("/emicheck");
    } catch (error) {
      toast.error(error.message, {
        position: "top-center",
        theme: "dark",
      });
    }
  };

  return (
    <>
      {/* {loading ? (
        <PuffLoader
          className="Loader"
          color={"#e60909"}
          loading={loading}
          size={100}
        />
      ) : ( */}
      <div className="container mt-5">
        <div className="d-flex justify-content-center h-100">
          <div className="card bg-dark col-6">
            <div className="card-header ">
              <h3 className="login-name text-light">
                <b>PropertyDetails</b>
              </h3>
            </div>
            <div className="card-body" style={{ backgroundColor: "#b30000" }}>
              <form onSubmit={handleOnSubmit}>
                <div className="input-group form-group text-light ">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faMapMarker}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Address</b>
                      </label>
                      <OverlayTrigger
                        trigger="hover"
                        placement="right"
                        overlay={customPopover}
                      >
                        <FontAwesomeIcon
                          className="fa ms-2"
                          icon={faCircleInfo}
                          style={{ fontSize: "1rem", color: "black" }}
                        ></FontAwesomeIcon>
                      </OverlayTrigger>
                    </span>
                  </div>
                  <input
                    type="text"
                    name="address"
                    aria-label="address"
                    className="form-control ms-2"
                    placeholder="Enter your address"
                    value={propertyDetails.address}
                    onChange={handleChange}
                  />
                </div>
                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faAreaChart}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Sq.Yard</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="text"
                    className="form-control ms-2"
                    name="area"
                    aria-label="area"
                    placeholder="Enter Area in square Yards"
                    value={propertyDetails.area}
                    onChange={handleChange}
                  />
                </div>
                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faMapPin}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Pincode</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    className="form-control ms-2"
                    name="pincode"
                    aria-label="Pincode"
                    placeholder="Pincode"
                    value={propertyDetails.pincode}
                    onChange={handleChange}
                  />
                </div>

                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faQuestion}
                        style={{
                          fontSize: "1.75em",
                          color: "black",
                          width: "30px",
                        }}
                      ></FontAwesomeIcon>
                      <label className="text-dark  ">
                        <b>Status</b>
                      </label>
                    </span>
                  </div>{" "}
                  <select
                    className="ms-2"
                    name="purposeOfLoan"
                    id="Purpose Of Loan"
                    value={propertyDetails.purposeOfLoan}
                    onChange={handleChange}
                    style={{ width: "442px ", borderRadius: "7px" }}
                  >
                    <option value="Purpose Of Loan" disabled>
                      Purpose Of Loan {""}
                    </option>
                    <option value="UnderConstruction">UnderConstruction</option>
                    <option value="Ready To Move">Ready To Move</option>
                    <option value="Home Renovation">Home Renovation</option>
                  </select>{" "}
                </div>
                <div className="form-group">
                  <Link
                    type="submit"
                    to="/"
                    className="btn float-right  btn-dark mt-4 px-4  "
                  >
                    Back
                  </Link>
                  <button
                    type="submit"
                    aria-label="Submit"
                    name="Submit"
                    value="Submit"
                    className="btn float-right  btn-dark mt-4 px-4 "
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      {/* )} */}
    </>
  );
}

export default PropertyDetails;
