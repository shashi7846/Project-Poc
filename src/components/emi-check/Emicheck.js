import React, { useState, useEffect } from "react";
import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCalendar,
  faPercent,
  faInr,
} from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-router-dom";
import { emiCalculator, emicheckValidation } from "../validators";
import { PuffLoader } from "react-spinners";
import { errorToast } from "../Toast";

const defaultData = {
  loanAmount: "",
  netIncome: "",
  tenure: "",
  price: "",
  interest: "9.5",
};

function Emicheck() {
  const [emicheck, setEmiCheck] = useState(defaultData);
  const [show, setShow] = useState(false);
  const [eligibility, setEligibility] = useState({
    loanAmount: 0,
    emi: 0,
    maxLoanAmount: 0,
    maxEmi: 0,
  });

  const [loading, setLoading] = useState(false);
  useEffect(() => {
    setLoading(true);
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, []);

  // let navigate = useNavigate();
  const handleChange = (e) =>
    setEmiCheck({ ...emicheck, [e.target.name]: e.target.value });
  const handleClose = () => setShow(false);

  const handleOnSubmit = (e) => {
    e.preventDefault();
    const errors = emicheckValidation(emicheck);
    if (errors.length) {
      errorToast(errors[0]);
      return;
    }
    const { loanAmount, netIncome, tenure, interest } = emicheck;
    const emi = emiCalculator(loanAmount, interest, tenure);
    const maxLoanAmount = Math.floor(((+netIncome * 0.5) / +emi) * loanAmount);
    const maxEmi = emiCalculator(maxLoanAmount, interest, tenure);
    setEligibility({ loanAmount, emi, maxLoanAmount, maxEmi });
    setShow(true);
  };

  return (
    <>
      {/* {loading ? (
        <PuffLoader
          className="Loader"
          color={'#e60909'}
          loading={loading}
          size={100}
        />
      ) : ( */}
      <div className="container mt-5">
        <div className="d-flex justify-content-center h-100">
          <div className="card bg-dark col-6">
            <div className="card-header ">
              <h3 className="login-name text-light">
                <b>Check Emi</b>
              </h3>
            </div>
            <div className="card-body" style={{ backgroundColor: "#b30000" }}>
              <form onSubmit={handleOnSubmit}>
                <div className="input-group form-group mt-3 mb-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      />{" "}
                      <label className="text-dark ms-2">
                        <b>PropertyPrice</b>
                      </label>
                    </span>
                  </div>

                  <input
                    type="number"
                    className="form-control ms-2"
                    name="price"
                    placeholder="Enter property price"
                    value={emicheck.price}
                    onChange={handleChange}
                  />
                </div>
                <div className="input-group form-group">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>LoanAmount</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    className="form-control ms-2"
                    name="loanAmount"
                    value={emicheck.loanAmount}
                    onChange={handleChange}
                    placeholder="Enter Loan Amount"
                  />
                </div>
                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Income P/M</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    className="form-control ms-2"
                    name="netIncome"
                    value={emicheck.netIncome}
                    onChange={handleChange}
                    placeholder="Net Income Per Month"
                  />
                </div>

                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faCalendar}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>LoanTenure</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    min={12}
                    max={120}
                    step={12}
                    className="form-control ms-2"
                    name="tenure"
                    value={emicheck.tenure}
                    onChange={handleChange}
                    placeholder=" Select Loan Tenure"
                  />
                </div>
                <div>
                  <label htmlFor="loan-amount" className="form-label"></label>
                  <input
                    type="range"
                    className="form-range mt-2 mb-3"
                    name="tenure"
                    min={12}
                    max={120}
                    step={12}
                    value={emicheck.tenure}
                    onChange={handleChange}
                  ></input>
                </div>

                <div className="input-group form-group">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faPercent}
                        style={{ fontSize: "1.75em", color: "black" }}
                      />{" "}
                      <label className="text-dark ms-2">
                        <b>InterestRate</b>
                      </label>
                    </span>
                  </div>

                  <input
                    type="number"
                    aria-label="interest"
                    className="form-control ms-2"
                    value={emicheck.interest}
                    placeholder="Rate Of Interest"
                    disabled
                  />
                </div>
                <div className="form-group mt-2">
                  <Link
                    type="submit"
                    to="/Propertydetails"
                    className="btn float-right  btn-dark mt-2 px-4 "
                  >
                    <b>Back</b>
                  </Link>
                  <Link
                    type="submit"
                    to="/"
                    className="btn float-right  btn-dark mt-2 px-4"
                  >
                    <b>Close</b>
                  </Link>
                  <button
                    type="submit"
                    name="Submit"
                    aria-label="Submit"
                    value="Submit"
                    className="btn float-right  btn-dark mt-2 px-4"
                  >
                    <b>Check Eligibility</b>
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      {/* )} */}

      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Loan Eligibility</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          {/* Emi For Entered Amount will be<b>₹ {eligibility.loanAmount}</b> at EMI
          <b> ₹ {eligibility.emi}</b> */}
          <br /> <h4>You are Eligible for a Maximum loan of</h4> <br />
          <b>₹ {eligibility.maxLoanAmount} </b>
          at EMI <b>₹ {eligibility.maxEmi}</b>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={handleClose}>
            <Link to="/confirmpage" style={{ color: "white" }}>
              {" "}
              Proceed
            </Link>
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}

export default Emicheck;
