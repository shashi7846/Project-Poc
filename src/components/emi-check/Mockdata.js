export const mockLoginData = {
  PropertyPrice: "10000000",
  LoanAmount: "100000",
  netIncome:"80000",
};
