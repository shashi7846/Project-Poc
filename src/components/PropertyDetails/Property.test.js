import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  renderComponent,
  renderComponentwithRouterPath,
} from "../../utils/test_helpers";
import { mockLoginData } from "./Mockdata";

import PropertyDetails from "./PropertyDetails";
import Emicheck from "../emi-check/Emicheck";

const user = userEvent.setup();
describe("property page", () => {
  test("should render Intial page", async () => {
    render(renderComponent(<PropertyDetails />));
    const pincode = screen.getByPlaceholderText("Pincode");
    await user.type(pincode, mockLoginData.pincode);
    expect(pincode).toHaveAttribute("type", "number");
    expect(pincode).toHaveValue(mockLoginData.number);
    // const Heading = screen.getByText(/address/i);
    // expect(Heading[0]).toBeInTheDocument();

    expect(screen.getByText("PropertyDetails")).toBeInTheDocument();
  });
  test("should render Intial paged", async () => {
    render(renderComponent(<PropertyDetails />));
    const pincode = screen.getByPlaceholderText("Pincode");
    await user.type(pincode, mockLoginData.pincode);
    expect(pincode).toHaveAttribute("type", "number");
    expect(pincode).toHaveValue(mockLoginData.number);
    // const Heading = screen.getByText(/address/i);
    // expect(Heading[0]).toBeInTheDocument();

    const Yards1 = screen.getByRole("textbox", { name: "area" });
    await user.type(Yards1, mockLoginData.area);
    expect(Yards1).toHaveValue(mockLoginData.area);

    const Address1 = screen.getByRole("textbox", { name: "address" });
    await user.type(Address1, mockLoginData.address);
    expect(Address1).toHaveValue(mockLoginData.address);
  });

  test("msw testing", async () => {
    renderComponentwithRouterPath(<PropertyDetails />, "/propertydetails", [
      { path: "/emicheck", element: <Emicheck /> },
    ]);

    const pincode = screen.getByPlaceholderText("Pincode");

    const Yards1 = screen.getByRole("textbox", { name: "area" });
    const Address1 = screen.getByRole("textbox", { name: "address" });

    const button = screen.getByRole("button", { name: "Submit" });
    await user.type(pincode, mockLoginData.pincode);

    await user.type(Yards1, mockLoginData.area);
    await user.type(Address1, mockLoginData.address);

    await user.click(button);
    waitFor(() => {
      expect(screen.getByText("Check Emi")).toBeInTheDocument();
    });
  });

  test("render properties", async () => {
    render(renderComponent(<PropertyDetails />));
    const properties = await screen.findAllByRole("textbox");
    expect(properties).toHaveLength(2);
  });
  test("Matching the snapshot of the propertydetails page", async () => {
    const { asFragment } = render(renderComponent(<PropertyDetails />));
    expect(asFragment()).toMatchSnapshot();
  });
});
