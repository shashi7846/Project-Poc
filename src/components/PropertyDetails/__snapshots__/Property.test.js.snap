// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`property page Matching the snapshot of the propertydetails page 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="card bg-dark col-6"
      >
        <div
          class="card-header "
        >
          <h3
            class="login-name text-light"
          >
            <b>
              PropertyDetails
            </b>
          </h3>
        </div>
        <div
          class="card-body"
          style="background-color: rgb(179, 0, 0);"
        >
          <form>
            <div
              class="input-group form-group text-light "
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-location-pin fa-beat-fade"
                    data-icon="location-pin"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Address
                    </b>
                  </label>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-circle-info fa ms-2"
                    data-icon="circle-info"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1rem; color: black;"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </div>
              <input
                aria-label="address"
                class="form-control ms-2"
                name="address"
                placeholder="Enter your address"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-chart-area fa-beat-fade"
                    data-icon="chart-area"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M64 64c0-17.7-14.3-32-32-32S0 46.3 0 64V400c0 44.2 35.8 80 80 80H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H80c-8.8 0-16-7.2-16-16V64zm96 288H448c17.7 0 32-14.3 32-32V251.8c0-7.6-2.7-15-7.7-20.8l-65.8-76.8c-12.1-14.2-33.7-15-46.9-1.8l-21 21c-10 10-26.4 9.2-35.4-1.6l-39.2-47c-12.6-15.1-35.7-15.4-48.7-.6L135.9 215c-5.1 5.8-7.9 13.3-7.9 21.1v84c0 17.7 14.3 32 32 32z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Sq.Yard
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="area"
                class="form-control ms-2"
                name="area"
                placeholder="Enter Area in square Yards"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-map-pin fa-beat-fade"
                    data-icon="map-pin"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M320 144c0 79.5-64.5 144-144 144S32 223.5 32 144S96.5 0 176 0s144 64.5 144 144zM176 80c8.8 0 16-7.2 16-16s-7.2-16-16-16c-53 0-96 43-96 96c0 8.8 7.2 16 16 16s16-7.2 16-16c0-35.3 28.7-64 64-64zM144 480V317.1c10.4 1.9 21.1 2.9 32 2.9s21.6-1 32-2.9V480c0 17.7-14.3 32-32 32s-32-14.3-32-32z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Pincode
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="Pincode"
                class="form-control ms-2"
                name="pincode"
                placeholder="Pincode"
                type="number"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-question fa-beat-fade"
                    data-icon="question"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black; width: 30px;"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 96c-17.7 0-32 14.3-32 32s-14.3 32-32 32s-32-14.3-32-32C0 75 43 32 96 32h97c70.1 0 127 56.9 127 127c0 52.4-32.2 99.4-81 118.4l-63 24.5 0 18.1c0 17.7-14.3 32-32 32s-32-14.3-32-32V301.9c0-26.4 16.2-50.1 40.8-59.6l63-24.5C240 208.3 256 185 256 159c0-34.8-28.2-63-63-63H96zm48 384c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark  "
                  >
                    <b>
                      Status
                    </b>
                  </label>
                </span>
              </div>
               
              <select
                class="ms-2"
                id="Purpose Of Loan"
                name="purposeOfLoan"
                style="width: 442px; border-radius: 7px;"
              >
                <option
                  disabled=""
                  value="Purpose Of Loan"
                >
                  Purpose Of Loan 
                </option>
                <option
                  value="UnderConstruction"
                >
                  UnderConstruction
                </option>
                <option
                  value="Ready To Move"
                >
                  Ready To Move
                </option>
                <option
                  value="Home Renovation"
                >
                  Home Renovation
                </option>
              </select>
               
            </div>
            <div
              class="form-group"
            >
              <a
                class="btn float-right  btn-dark mt-4 px-4  "
                href="/"
                type="submit"
              >
                Back
              </a>
              <button
                aria-label="Submit"
                class="btn float-right  btn-dark mt-4 px-4 "
                name="Submit"
                type="submit"
                value="Submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
